woocommerce/woocommerce-ios
# [Composite Products] Add Component Settings view and view model

Part of #8956


## Description

This adds a new Component Settings view for displaying read-only component settings. It displays the provided settings:

* If the image is `nil`, it is hidden (like the read-only product details screen).
* If the description is empty, it is hidden (like the read-only product details screen).
* If the options list is empty, it shows a "No options selected" placeholder.

## Testing instructions

This view is not yet used in the app, so you can check the SwiftUI previews or screenshots below to see how it looks.

## Screenshots


With images and description|Without images and description|Without options
-|-|-
![Screenshot 2023-03-24 at 1 56 32 PM](https://user-images.githubusercontent.com/8658164/227571141-c86bd48a-8312-43e7-92b1-9aa76a4d6e45.png)|![Screenshot 2023-03-24 at 3 28 42 PM](https://user-images.githubusercontent.com/8658164/227571157-d5bf9389-b15d-4eef-aa03-2c577cfac879.png)|![Screenshot 2023-03-24 at 5 30 14 PM](https://user-images.githubusercontent.com/8658164/227598464-efdb1410-cda5-4783-ad67-155d6037d644.png)


---
